Bash-completion is written with FHS in mind where completion scripts
all get added to /usr/share/bash-completion/completions and are picked
up by the dynamic completion loader from there---whether they are
part of bash-completion or installed by another package.

On Guix systems, we want not only to search within bash-completion's own
directory, but also in the user's profile and guix home profile, as well
as in the system profile.
This is what this patch does.


--- a/bash_completion
+++ b/bash_completion
@@ -3192,7 +3192,16 @@
             dirs+=("${dir%/*}/share/bash-completion/completions")
     done
 
-    # 4) From XDG_DATA_DIRS or system dirs (e.g. /usr/share, /usr/local/share):
+    # 4) Guix-specific
+    dirs+=(
+        "$HOME/.guix-profile/share/bash-completion/completions"
+        "$HOME/.guix-profile/etc/bash_completion.d"
+        "$HOME/.guix-home/profile/share/bash-completion/completions"
+        "$HOME/.guix-home/profile/etc/bash_completion.d"
+        "/run/current-system/profile/share/bash-completion/completions"
+        "/run/current-system/profile/etc/bash_completion.d" )
+
+    # 5) From XDG_DATA_DIRS or system dirs (e.g. /usr/share, /usr/local/share):
     # Completions in the system data dirs.
     _comp_split -F : paths "${XDG_DATA_DIRS:-/usr/local/share:/usr/share}" &&
         dirs+=("${paths[@]/%//bash-completion/completions}")
